File: Gestion-De-Stock-BackEnd/src/main/java/com/hassan/gestiondestock/dto/ArticleDto.java
package com.hassan.gestiondestock.dto;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.Lob;
import java.math.BigDecimal;
import java.util.List;

@Builder
@Data
@AllArgsConstructor
@NoArgsConstructor
public class ArticleDto {

    private Integer id;

    private String  codeArticle;

    private String  designation;

    private BigDecimal prixUnitaireHt;

    private BigDecimal  prixUnitaireTtc;

    private BigDecimal  touxTva;

    private byte[] photo;

    private String  nomPhoto;

    private CategoryDTO categoryId;

    @JsonIgnore
    private List<mvtDto> mvtDeStocks;

    private String idEntreprise;


    @JsonIgnore
    private List<LigneVenteDto> ligneVentes;

    @JsonIgnore
    private List<ligneCommandeFournisseurDto> ligneCommandeFournisseurs;

    @JsonIgnore
    private List<ligneCommandeClientsDto> ligneCommandeClientsList;

}

File: gestionDeStock-Front-End/src/app/DTO/articleDTO.ts
import {categoryDTO} from './categoryDTO';
import {ImageUploadModule} from './image-upload.module';

export class articleDTO {
  id: number;
  codeArticle: string;
  designation:string;
  prixUnitaireHt: number;
  touxTva: number;
  photo:ImageUploadModule=null;
  prixUnitaireTtc: number;
  categoryId: categoryDTO;
  idEntreprise:string;
  quantite:number;


  constructor() {
    this.codeArticle = "";
    this.designation = "";
    this.prixUnitaireHt = null;
    this.touxTva = null;
    this.photo = null;
    this.prixUnitaireTtc = null;
    this.categoryId = null;
    this.idEntreprise = "";
    this.quantite=null;
  }
}
